File: apps/admin/domain/types/branch.ts
/**
 * Branch & Staff Domain Types
 *
 * Type definitions for the branch and staff domains.
 *
 * @module domain/types/branch
 */

// ─── Branch Entity ───────────────────────────────────────────────────
export interface Branch {
  readonly id: string;
  readonly store_id: string;
  name: string;
  address: string;
  phone: string | null;
  is_main: boolean;
  is_active: boolean;
  readonly created_at: string;
  readonly updated_at: string;
  // Audit fields
  readonly created_by: string | null;
  readonly created_at_branch_id: string | null;
  readonly updated_by: string | null;
  readonly updated_at_branch_id: string | null;
}

export interface BranchWithStaffCount extends Branch {
  staff_count?: number;
}

// ─── Staff Entity ────────────────────────────────────────────────────
export type StaffRole = 'super_admin' | 'admin' | 'manager' | 'staff';

export interface Staff {
  readonly id: string;
  readonly store_id: string;
  readonly branch_id: string;
  readonly user_id: string | null; // links to auth.users
  name: string;
  email: string;
  phone: string | null;
  role: StaffRole;
  is_active: boolean;
  readonly created_at: string;
  readonly updated_at: string;
  // Audit fields
  readonly created_by: string | null;
  readonly created_at_branch_id: string | null;
  readonly updated_by: string | null;
  readonly updated_at_branch_id: string | null;
}

export interface StaffWithBranch extends Staff {
  branch?: Pick<Branch, 'id' | 'name'>;
}

// ─── Branch DTOs ─────────────────────────────────────────────────────
export interface CreateBranchDTO {
  name: string;
  address: string;
  phone?: string;
  is_main?: boolean;
  is_active?: boolean;
  store_id: string;
}

export interface UpdateBranchDTO {
  name?: string;
  address?: string;
  phone?: string | null;
  is_main?: boolean;
  is_active?: boolean;
}

// ─── Staff DTOs ──────────────────────────────────────────────────────
export interface CreateStaffDTO {
  name: string;
  email: string;
  password: string; // for Supabase Auth user creation
  phone?: string;
  role: StaffRole;
  branch_id: string;
  is_active?: boolean;
  store_id: string;
}

export interface UpdateStaffDTO {
  name?: string;
  email?: string;
  phone?: string | null;
  role?: StaffRole;
  branch_id?: string;
  is_active?: boolean;
}

// ─── Search Params ───────────────────────────────────────────────────
export interface BranchSearchParams {
  query?: string;
  is_active?: boolean;
  limit?: number;
  offset?: number;
}

export interface StaffSearchParams {
  query?: string;
  branch_id?: string;
  role?: StaffRole;
  is_active?: boolean;
  limit?: number;
  offset?: number;
}

File: apps/mobile/lib/features/branches/models/branch.dart
class Branch {
  final String id;
  final String storeId;
  final String name;
  final String address;
  final String? phone;
  final bool isMain;
  final bool isActive;
  final String createdAt;
  final String updatedAt;
  final String? createdBy;
  final String? createdAtBranchId;
  final String? updatedBy;
  final String? updatedAtBranchId;
  final int? staffCount;

  Branch({
    required this.id,
    required this.storeId,
    required this.name,
    required this.address,
    this.phone,
    this.isMain = false,
    this.isActive = true,
    required this.createdAt,
    required this.updatedAt,
    this.createdBy,
    this.createdAtBranchId,
    this.updatedBy,
    this.updatedAtBranchId,
    this.staffCount,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'] as String,
      storeId: json['store_id'] as String,
      name: json['name'] as String,
      address: json['address'] as String,
      phone: json['phone'] as String?,
      isMain: json['is_main'] as bool? ?? false,
      isActive: json['is_active'] as bool? ?? true,
      createdAt: json['created_at'] as String? ?? '',
      updatedAt: json['updated_at'] as String? ?? '',
      createdBy: json['created_by'] as String?,
      createdAtBranchId: json['created_at_branch_id'] as String?,
      updatedBy: json['updated_by'] as String?,
      updatedAtBranchId: json['updated_at_branch_id'] as String?,
      staffCount: json['staff_count'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'store_id': storeId,
      'name': name,
      'address': address,
      'phone': phone,
      'is_main': isMain,
      'is_active': isActive,
      'created_at': createdAt,
      'updated_at': updatedAt,
      'created_by': createdBy,
      'created_at_branch_id': createdAtBranchId,
      'updated_by': updatedBy,
      'updated_at_branch_id': updatedAtBranchId,
      'staff_count': staffCount,
    };
  }
}
